src/state/routing/types.ts

```diff
@@ -187,6 +193,21 @@ export class ClassicTrade extends Trade<Currency, Currency, TradeType> {
     this.requestId = requestId
     this.quoteMethod = quoteMethod
     this.approveInfo = approveInfo
+    this.inputTax = inputTax
+    this.outputTax = outputTax
+  }
+
+  public get totalTaxRate(): Percent {
+    return this.inputTax.add(new Fraction(ONE).subtract(this.inputTax).multiply(this.outputTax))
```

Comment:
I'll add a comment, this is to handle for swapping from FOT to FOT. The output tax is only applied to what's left of the amountIn after the input tax has already been applied

So the total tax rate is `input_tax + ((1 - input_tax) * output_tax)` rather than `input_tax + output_tax`

Reply:
hmm so i believe this assumes X% decrease in amount in means a X% decrease in amount out, which isnt right, but i also think this case is rare enough where we don't have to be ultra precise about it

i'd vote for doing the simpler but looser input_tax + output_tax until we can actually amend the `amountIn` calculated by the sdk, what do u think?

Reply:
sounds good to me!